File: Extra/decrescente.c
#include <stdio.h>

void ordenar_decrescente(int * arr, int tamanho){
    for(int i = 1; i < tamanho; i++){
        int temp = arr[i];
        int j = i - 1;

        while(j >= 0 && arr[j] < temp){
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }
}

int main(){

    int arr[5] = {3,1,4,8,5};
    int tamanho = sizeof(arr) / sizeof(arr[0]);

    ordenar_decrescente(arr, tamanho);

    for(int i = 0; i < tamanho; i++){
        printf("%d ", arr[i]);
    }

    return 0;
}

File: Extra/decrescente.py
def ordenar_decrescente(arr, tamanho):
    for i in range(1, tamanho):
        temp = arr[i]
        j = i - 1 

        while j >= 0 and arr[j] < temp:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = temp

arr = [3,1,4,8,5]

ordenar_decrescente(arr, len(arr))

print(arr)